Remotion/Core/Core/Collections/ReadOnlyListExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using Remotion.Utilities;
+
+namespace Remotion.Collections
+{
+  /// <summary>
+  /// Provides useful extension methods for <see cref="IReadOnlyList{T}"/>.
+  /// </summary>
+  public static class ReadOnlyListExtensions
+  {
+    /// <summary>
+    /// Searches for the specified <paramref name="value"/> and returns the index of its
+    /// first occurence in <paramref name="list"/>.
+    /// </summary>
+    /// <param name="list">The list to search <paramref name="value"/> in.</param>
```

Comment:
```suggestion
    /// <param name="list">The list to search <paramref name="value"/> in. Must not be <see langword="null" /></param>
```

(we have a live template for that phrase)